rafthttp/stream.go

```diff
@@ -309,24 +309,20 @@ func (cr *streamReader) isWorking() bool {
 
 func (cr *streamReader) dial() (io.ReadCloser, error) {
 	cr.mu.Lock()
-	u := cr.u
+	u := cr.urls[rand.Intn(len(cr.urls))]
```

Comment:
same here. it makes more sense to have a url picker used by both stream and pipeline, which has a few urls and can pick one.